c_src/quicer_listener.c

```diff
@@ -473,13 +474,19 @@ close_listener1(ErlNifEnv *env,
   enif_mutex_lock(l_ctx->lock);
   HQUIC l = l_ctx->Listener;
   l_ctx->Listener = NULL;
+
+  if (l_ctx->is_closed)
+  {
+    ret = ERROR_TUPLE_2(ATOM_CLOSED);
+  }
   l_ctx->is_closed = TRUE;
   enif_mutex_unlock(l_ctx->lock);
 
   // It is safe to close it without holding the lock
   // This also ensures no ongoing listener callbacks
-  // This is a blocking call. @TODO have async version or use dirty scheduler
```

Comment:
was the *TODO* removed on purpose or by accident? (as it's not related to this change)

Reply:
We could use ListenerStop for async calls instead and keep ListenerCLose sync. 
maybe it is a good time to add ListenerStop support. 
I will update the PR.